### PR TITLE
Refactor OpAMP bridge logging and configuration

### PR DESCRIPTION
**Pull request by jaronoff97**
As a side quest from #1368, this PR changes how config is loaded in to the bridge as well as the logging structure.

This has some breaking changes around the config. This is okay because currently no one is running the bridge (AFAIK!)

`components_allowed` => `componentsAllowed`
:x: `protocol` which is now inferred from endpoint
capabilities `[]string` => `map[Capability]bool` for enhanced configuration validation

### CODE REVIEW

**Review by srikanthccv — APPROVED** [review]
LGTM

**Comment by avadhut123pisal** [comment]
@jaronoff97 What is your view on https://github.com/open-telemetry/opentelemetry-operator/pull/1559#discussion_r1347470831 ? Because, that would require change in the OpAMP bridge code as well as in the Operator.

**Comment by jaronoff97** [comment]
@avadhut123pisal i like the recommendation, ill incorporate it here

**Comment by srikanthccv** [comment]
My reasoning behind using typed struct is as the bridge implementation improves, we will find ourselves checking if a capability exists before performing certain actions (for instance, AcceptConnectionSettings). 

I find it better to do some like `if b.Capabilities.AcceptsRemoteConfig` than enumerating the options and comparing.

**Comment by jaronoff97** [comment]
We are able to do that with a map check with an enum as well. I'm trying to avoid needing to do an if on every struct field for validation or defaulting.

**Comment by srikanthccv** [comment]
I am more biased towards the end user's POV/experience than what needs to be done in our codebase. The enums/strings are prone to typos. The struct makes it clear about the fixed set of capabilities that the user can enable or disable and works nicely with the IDE yaml schema generation and validation. I don't have a strong opinion. I am fine with either of the approaches.

**Comment by jaronoff97** [comment]
Using a map is the same user experience while also being more ergonomic for use in go by creating constants for each capability.

**Review by avadhut123pisal — APPROVED** [review]
LGTM :)

File: cmd/operator-opamp-bridge/agent/agent.go [hunk 1]
```diff
@@ -64,7 +66,7 @@ func NewAgent(logger types.Logger, applier operator.ConfigApplier, config config
 		opampClient:         opampClient,
 	}
 
-	agent.logger.Debugf("Agent created, id=%v, type=%s, version=%s.",
+	agent.logger.V(3).Info("Agent created, id=%v, type=%s, version=%s.",
```
[the remark this answers is not in the document]
Comment (jaronoff97) [reply cmd/operator-opamp-bridge/agent/agent.go:69]: ah yes, i should do this. this was copied from the example code and previously that type of kv thing wasn't supported :)

File: cmd/operator-opamp-bridge/main.go [hunk 2]
```diff
@@ -18,37 +18,38 @@ import (
...
 	l := config.GetLogger()
-	cliConf, err := config.ParseCLI(l.WithName("cli-config"))
+
+	flagSet := config.GetFlagSet(pflag.ExitOnError)
+	err := flagSet.Parse(os.Args)
 	if err != nil {
-		l.Error(err, "unable to load ")
-		os.Exit(1)
+		l.Error(err, "Unable to load flags")
+		return
```
[the remark this answers is not in the document]
Comment (jaronoff97) [reply cmd/operator-opamp-bridge/main.go:35]: yeah, we should still return 1. fixing.

File: cmd/operator-opamp-bridge/config/config.go [hunk 3]
```diff
@@ -16,43 +16,99 @@ package config
...
+	AcceptsRestartCommand          Capability = "AcceptsRestartCommand"
+	ReportsHealth                  Capability = "ReportsHealth"
+	ReportsRemoteConfig            Capability = "ReportsRemoteConfig"
 )
 
 type Config struct {
-	Endpoint     string   `yaml:"endpoint"`
-	Protocol     string   `yaml:"protocol"`
-	Capabilities []string `yaml:"capabilities"`
+	// KubeConfigFilePath empty if in cluster configuration is in use
```
Comment (gdfast) [inline cmd/operator-opamp-bridge/config/config.go:87]: Something about this wording is a little confusing, maybe you have an idea of how to clarify. Or maybe there's an extra "in" in here

File: cmd/operator-opamp-bridge/config/config.go [hunk 4]
```diff
@@ -16,43 +16,99 @@ package config
...
+	return nil
+}
+
+func GetLogger() logr.Logger {
+	return zap.New(zap.UseFlagOptions(&zapCmdLineOpts))
+}
+
+type Capability string
+
+const (
```
Comment (gdfast) [inline cmd/operator-opamp-bridge/config/config.go:69]: nit: is there an approach to the order of these consts? I think alpha-ordering might be easier to look at and review

**Review by gdfast — APPROVED** [review]
LGTM!

File: cmd/operator-opamp-bridge/config/config.go (hunk 4 above)
Comment (jaronoff97) [reply cmd/operator-opamp-bridge/config/config.go:69]: this came from the upstream opamp-go ordering, but agreed... it's a bit odd.

File: cmd/operator-opamp-bridge/config/config.go (hunk 3 above)
Comment (jaronoff97) [reply cmd/operator-opamp-bridge/config/config.go:87]: yeah i can clear that up

**Review by gdfast — APPROVED** [review]

File: .chloggen/refactor-bridge-config.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+# One or more tracking issues related to the change
+issues: [1368]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  `components_allowed` => `componentsAllowed`
+  :x: `protocol` which is now inferred from endpoint
```
Comment (pavolloffay) [inline .chloggen/refactor-bridge-config.yaml:18]: what does `:x:` mean?  Does it mean that is has been removed?

**Review by pavolloffay — APPROVED** [review]

File: .chloggen/refactor-bridge-config.yaml (hunk 5 above)
Comment (jaronoff97) [reply .chloggen/refactor-bridge-config.yaml:18]: Yes

**Pull request merged**